tests/MongoDB.Driver.Tests/Specifications/uuid/prose-tests/ImplicitEncodingTests.cs

```diff
@@ -0,0 +1,190 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Bson.Serialization.Attributes;
+using MongoDB.Bson.TestHelpers;
+using System;
+using System.Linq;
+using Xunit;
+
+namespace MongoDB.Driver.Tests.Specifications.uuid.prose_tests
+{
+    public class ImplicitEncodingTests
+    {
+        [Fact]
+        public void ClassWithGuidIdUsingCSharpLegacyRepresentation_tests()
```

Comment:
These tests look pretty different than the prose they come from but it's the closest C# driver equivalent to what the prose tests are talking about.